core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,943 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Arrays;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.catalog.ViewCatalog;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.exceptions.NoSuchViewException;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Types;
+import org.assertj.core.api.Assumptions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public abstract class ViewCatalogTests<C extends ViewCatalog & SupportsNamespaces> {
+  protected static final Schema SCHEMA =
+      new Schema(
+          1,
+          required(3, "id", Types.IntegerType.get(), "unique ID"),
+          required(4, "data", Types.StringType.get()));
+
+  private static final Schema OTHER_SCHEMA =
+      new Schema(2, required(1, "some_id", Types.IntegerType.get()));
+
+  protected abstract C catalog();
+
+  protected abstract Catalog tableCatalog();
+
+  protected boolean requiresNamespaceCreate() {
+    return false;
+  }
+
+  @Test
+  public void basicCreateView() {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+
+    View view =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withQuery("spark", "select * from ns.tbl")
+            .create();
+
+    assertThat(view).isNotNull();
+    assertThat(catalog().viewExists(identifier)).as("View should exist").isTrue();
+
+    // validate view settings
+    assertThat(view.name()).isEqualTo(catalog().name() + "." + identifier);
+    assertThat(view.properties()).isEmpty();
+    assertThat(view.history())
+        .hasSize(1)
+        .first()
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(1);
+    assertThat(view.schemas()).hasSize(1).containsKey(SCHEMA.schemaId());
```

Comment:
Why does this schema ID match the constant's ID? It should be reassigned in the `ViewMetadata` builder.

Reply:
the schema ID of `SCHEMA` / `OTHER_SCHEMA` is hardcoded for easier testing, so schema ID reassigning doesn't happen currently

Reply:
It shouldn't match so that this tests that the schema ID is reassigned. Otherwise, this isn't correctly testing the guarantees. Reading this test, you'd think that the API is designed to preserve the `schemaId` from the incoming schema, which is not true. This should use the wrong schema ID for the constant and assert that the assigned ID is `1`.

What's also concerning here is that the schema ID of the constant is `2`, which means ID assignment is probably incorrect.